compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -874,6 +874,7 @@ public BLangNode transform(UnionTypeDescriptorNode unionTypeDescriptorNode) {
             unionTypeNode.memberTypeNodes.add(createTypeNode(unionElement));
         }
 
+        bLangFiniteTypeNode.setPosition(unionTypeNode.pos);
```

Comment:
For now, this is ok I guess. But this should be set to the position covering the finite type elements. i.e., starting pos should be the first element's starting pos and the ending pos should be the last element's ending pos.